### PR TITLE
[FEATURE] Push Bullet Notifications

### PR DESCRIPTION
**Pull request by CapslockRSPS**
Provide a general summary of your changes in the **Title** above.

### Important
Mark with [x] to select. Leave as [ ] to unselect.
If possible; include a screenshot or gif of the change you've made

### Motivation and Context
Each item you can check should be described in the _Description_ section.

- [ ] Why is this change required? What problem does it solve?
- [ ] If it fixes an open issue, include the text `Closes issue #1` (where 1 would be the issue number) to your commit message.

### Types of changes
What types of changes does your code introduce? Check all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Cleanup (non-breaking change which cleans up the code)

### Final checklist
Go over all the following points and check all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help! 
Various areas of the codebase have been worked on by different people in recent years, so if you are unfamiliar with the general area you're working in, please feel free to chat with people who have experience in that area.

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have tested the functionality
- [ ] I have added tests for this functionality

Your pull request will be tested via Travis CI to automatically indicate that your changes do not prevent compilation.

If it reports back that there are problems, you go to [the Travis system](https://travis-ci.org/Parabot/Parabot) and check the log report for your pull request to see what the problem was.
If you add new code to fix a Travis building issue/problem, then take note that you need to check the next pull request in the Travis system.
Travis issue numbers are different from GitHub issue numbers.

### Description
Will be used in the new up coming notification system.

### CODE REVIEW

**Comment by CapslockRSPS** [comment]
Leaving a note that the url for the library needs to be changed/updated since that jar is from 2014.

**Comment by JKetelaar** [comment]
What if the user doesn't want to use PushBullet?

**Comment by CapslockRSPS** [comment]
It should only download it once it needs it.

File: src/main/java/org/parabot/core/lib/jpushbullet/JPushBullet.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    @Override
+    public File getJarFile() {
+        return new File(Directories.getCachePath(), "jpushbullet.jar");
+    }
+
+    @Override
+    public URL getDownloadLink() {
+        try {
+            return new URL("https://github.com/silk8192/jpushbullet/releases/download/1.0/jpushbullet-1.0.jar");
```
Comment (JKetelaar) [inline src/main/java/org/parabot/core/lib/jpushbullet/JPushBullet.java]: Needs to be updated to `Configuration.LIBRARIES_DOWNLOAD + "/JPushBullet"`

**Comment by JKetelaar** [comment]
Internal-API dependency must be updated to `1.4.5`.

File: src/main/java/org/parabot/core/lib/jpushbullet/JPushBullet.java (hunk 1 above)
Comment (JKetelaar) [inline src/main/java/org/parabot/core/lib/jpushbullet/JPushBullet.java]: Needs to be updated to `Configuration.LIBRARIES_DOWNLOAD + "/JPushBullet"`

**Pull request merged**